discord/interactions/addConfigCommand.ts

```diff
@@ -280,34 +283,47 @@ export const handleConfigConfirmCommand = async (
 
   const currentConfig = ongoingConfigurationsCache[interaction.guildId];
 
-  const alreadyDiscordServer = await DiscordServerRepository.findOneBy({
-    id: interaction.guildId,
-  });
+  const alreadyDiscordServerConfig =
+    await DiscordServerConfigRepository.findOneBy({
+      DiscordServerId: interaction.guildId,
+      starkyModuleConfig: currentConfig.moduleConfig,
+      discordRoleId: currentConfig.roleId,
+    });
 
-  const discordServer = alreadyDiscordServer || new DiscordServer();
-  discordServer.id = interaction.guildId;
+  const discordServerConfig =
+    alreadyDiscordServerConfig || new DiscordServerConfig();
+  discordServerConfig.DiscordServerId = interaction.guildId;
   if (
     currentConfig.network !== "mainnet" &&
     currentConfig.network !== "goerli"
   ) {
     throw new Error("Wrong network config");
   }
-  discordServer.starknetNetwork = currentConfig.network;
+  discordServerConfig.starknetNetwork = currentConfig.network;
   if (!currentConfig.roleId) {
     throw new Error("Wrong role config");
   }
-  discordServer.discordRoleId = currentConfig.roleId;
+  discordServerConfig.discordRoleId = currentConfig.roleId;
   if (
     !currentConfig.moduleType ||
     !(currentConfig.moduleType in starkyModules)
   ) {
     throw new Error("Wrong module config");
   }
-  discordServer.starkyModuleType = currentConfig.moduleType;
+  discordServerConfig.starkyModuleType = currentConfig.moduleType;
 
-  discordServer.starkyModuleConfig = currentConfig.moduleConfig || {};
+  discordServerConfig.starkyModuleConfig = currentConfig.moduleConfig || {};
 
-  await DiscordServerRepository.save(discordServer);
+  await DiscordServerConfigRepository.save(discordServerConfig);
+
+  const NewDiscordMembers = await DiscordMemberRepository.findBy({
+    starknetWalletAddress: Not(IsNull()),
+  });
+  for (let discordMember of NewDiscordMembers) {
+    discordMember.DiscordServerConfig = discordServerConfig;
+    discordMember.DiscordServerConfigId = discordServerConfig.id;
+    await DiscordMemberRepository.insert(discordMember);
+  }
```

Comment:
I don't understand this. It seems you're taking ALL the discord members with a starknet address and assigning them to this discordServerConfig ?